### ISSUE TITLE
Parent contentControl is not found on the event DocumentSelectionChanged

### ISSUE THREAD

**Issue by benjamin-almarza** [OfficeDev/office-js#2344]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

We add an event in the document that triggers when a contentControl is modified. Our issue is that in some documents (not all) the placeholder is null from the point of view of the addin.

## Your Environment
Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web
Host [Excel, Word, PowerPoint, etc.]: Word
Operating System: Windows 10 Pro
Browser (if using Office on the web): Chrome is up to date Version 96.0.4664.110 (Official Build) (64-bit)
 
## Expected behavior
On selecting inside a contentControl: we should be able to access its tag and id.

## Current behavior
On selecting inside a contentControl:  contentControl.isNullObject is true.

## Steps to reproduce
Upload the file [bug_file.docx](https://github.com/OfficeDev/office-js/files/7806235/bug_file.docx) to office.com

Run the following code in Office Word addin online
```javascript
const onPlaceHolderSelected = async () => {
  return Word.run(async context => {
    const range = context.document.getSelection();
    context.load(range, "text");

    const contentControl = range.parentContentControlOrNullObject;
    context.load(contentControl, "text, tag, id");

    await context.sync();

    const tag = contentControl.tag;
    if (contentControl.isNullObject) {
      return Promise.reject();
    }

    return { id: contentControl.id, tag: tag };
  });
}

Office.onReady(info => {
  Word.run(async () => {
    return Office.context.document.addHandlerAsync(Office.EventType.DocumentSelectionChanged, () => {
      onPlaceHolderSelected().then( value => {
        console.log(value);
      })
      .catch(() => {
        console.error("not a contentControl")
      });
    })
  })
});

```

Click with the mouse in the word "placeholder"

Ideally, the line "console.log(value);" should be run.

## Context
Our application relies heavily on contentControl without this piece it doesn't work.

**Comment by jingxuan0109** [comment]
Hi, I can not repro this issue. I am just wondering whether you can provide more details. 

![image](https://user-images.githubusercontent.com/86349902/148166822-573308a3-db96-4599-a101-f7ec12699cfb.png)

**Comment by benjamin-almarza** [comment]
Hi: I figured out something weird.

If we do the following: 
* create a document
* upload the document to office.com
* test as I explain

Then the issue happens as I described

But if we do the following: 
* create a document
* upload the document to office.com
* download the document
* re-upload the document to office.com

Then it doesn't fails
I can see that the document.xml inside the docx changes a little

I attached a brand new file that fails consistently when I upload it :
[test_file_2.docx](https://github.com/OfficeDev/office-js/files/7812830/Venkat_Test_Template_Simple.docx)

Please upload this and test

--------------
I attached the document.xml before and after:
[document_after.xml](https://github.com/OfficeDev/office-js/files/7813110/document_after.xml.txt)
[document_before.xml](https://github.com/OfficeDev/office-js/files/7813111/document_before.xml.txt)

The most significative change I can see is from this:

```xml
  <w:sdtContent>
                    <w:r>
```
to this:
```xml
  <w:sdtContent>
                    <w:r w:rsidRPr="6826A8E4" w:rsidR="73081A6C">
```

**Comment by jingxuan0109** [comment]
I am just wondering how you insert a content control. It seems I cannot get any content controls for you document. But it can be get for new inserted content control.

**Comment by benjamin-almarza** [comment]
1.- How we created the document 

We are creating programmatically the document using aspose
* Library: Aspose for Words
* Version: aspose-words-18.4-jdk16

This is the snippet of the code 
```ruby
#create a new content control
sdt = Rjb::import('com.aspose.words.StructuredDocumentTag').new(aspose_doc, sdt_type_plain_text, markup_level_inline)
#lock it to prevent from accidental deletion in MS-Word
sdt.setLockContentControl(true)
#add tag for identification
sdt.setTag(tag_text)
#set text for context control, font etc.
sdt.appendChild(run_node(display_text, fontName, fontSize, fontColor))
sdt.isShowingPlaceholderText(false)

#replace the default text message
sdt.getRange.replace("Click here to enter text.", "", false, false)
doc_builder.insertNode(sdt)
```

2.- The contentControls we have.
For every purpose, the contentControls we create work fine.

They look and behave as normal contentControls as you can see
![screenshot](https://user-images.githubusercontent.com/51302934/148341964-dbd94af8-e451-47dd-be36-d3fd0b74e07b.png)

Besides, one of our requirements is that we cannot control the documents received from our clients. If it is a validly created Docx with contentControls we should be able to interact with them.

3.- Document to test

I attached a document before and after downloading it from office.com. If you upload both to office.com and run the test I gave you,  in one the contentControl "contract name" will work and in the other not. This may be useful to test because they are practically the same file.

* [test_document_original.docx](https://github.com/OfficeDev/office-js/files/7819971/test_document_original.docx)
* [test_document_modified.docx](https://github.com/OfficeDev/office-js/files/7819970/test_document_modified.docx)

**Comment by jerryvaz** [comment]
@RuoyingLiang Did you get a chance to look into this case? We are pretty stuck and blocked on this issue.

**Comment by jingxuan0109** [comment]
I am just wondering whether you can try richtext content control.

**Comment by chiz-ms** [comment]
The issue is tracked in our backlog as ADO#5705306.

**Comment by benjamin-almarza** [comment]
The issue disappears when we change plain text content control to rich text.
(Thanks, this would help us for the newer created documents, but we need a solution for the old one )
